registry/auth/token/token.go

```diff
@@ -144,8 +147,14 @@ func (t *Token) Verify(verifyOpts VerifyOptions) error {
 
 	// Verify that the token is currently usable and not expired.
 	currentUnixTime := time.Now().Unix()
-	if !(t.Claims.NotBefore <= currentUnixTime && currentUnixTime <= t.Claims.Expiration) {
-		log.Errorf("token not to be used before %d or after %d - currently %d", t.Claims.NotBefore, t.Claims.Expiration, currentUnixTime)
+
+	if currentUnixTime >= t.Claims.Expiration+int64(Leeway.Seconds()) {
+		log.Errorf("token not to be used after %d - currently %d", t.Claims.Expiration, currentUnixTime)
+		return ErrInvalidToken
+	}
+
+	if currentUnixTime <= t.Claims.NotBefore-int64(Leeway.Seconds()) {
```

Comment:
I'd feel more comfortable if we converted to time value, did arithmetic, then use `Before`/`After`.